### PR TITLE
[c10d]add coalesce support for device types other than cuda

### PR DESCRIPTION
**Pull request by siju-samuel**
- This change allows easy extension of supported device types by updating the `COALESING_SUPPORTED_DTYPES`.
- 'has_coalescing_manager' method can be used in other places as well.

Fixes #ISSUE_NUMBER
cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @yf225 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @yanboliang @bsochack

### CODE REVIEW

**Comment by siju-samuel** [comment]
@wconstab @kwen2501 Could you lease help to review this change. TIA

File: torch/distributed/distributed_c10d.py
Comment (kwen2501) [inline torch/distributed/distributed_c10d.py:2235]: nit: typo? COALESING -> COALESCING
nit: dtype or device type? (the former is often understood as data type)

Comment (kwen2501) [reply torch/distributed/distributed_c10d.py:2235]: nit: prefix this variable with "_" ?

File: torch/distributed/distributed_c10d.py
Comment (kwen2501) [inline torch/distributed/distributed_c10d.py:2237]: nit: `has_coalescing_manager` --> `supports_coalescing_manager`

**Review by kwen2501 — APPROVED** [review]
LGTM overall. Approving to unblock. Do you mind addressing the comments before land? Thanks!

**Comment by kwen2501** [comment]
There is a similar PR:
https://github.com/pytorch/pytorch/pull/135338

It seems adding the property at backend level makes sense.

**Pull request closed without merging**
